frontend/testing/playwright/pages/DashboardPage.ts

```diff
@@ -15,6 +16,70 @@ export class DashboardPage extends BasePage {
   }
 
   public async clickOnCreateAppLink(): Promise<void> {
-    await this.page.getByRole('link', { name: this.textMock('dashboard.new_service') }).click();
+    await this.getLinkByTextKey('dashboard.new_service').click();
+  }
+
+  public async checkThatThereIsNoFavouriteAppInList(appName: string): Promise<void> {
+    // The .first() is added becuase the key is used two places; one in favourite list, and one in all applications list
+    await this.getMenuItemByTextKey('dashboard.unstar', { appName }).first().isHidden();
+  }
+
+  public async clickOnFavouriteApplication(appName: string): Promise<void> {
+    await this.getMenuItemByTextKey('dashboard.star', { appName }).click();
+  }
+
+  public async checkThatThereIsFavouriteAppInList(appName: string): Promise<void> {
+    await this.getMenuItemByTextKey('dashboard.star', { appName }).isVisible();
+  }
+
+  public async clickOnUnFavouriteApplicatin(appName: string): Promise<void> {
+    // The .first() is added becuase the key is used two places; one in favourite list, and one in all applications list
+    await this.getMenuItemByTextKey('dashboard.unstar', { appName }).first().click();
+  }
+
+  public async clickOnHeaderAvatar(): Promise<void> {
+    await this.page.getByAltText(this.textMock('shared.header_button_alt')).click();
+  }
+
+  public async clickOnAllApplications(): Promise<void> {
+    await this.getMenuItemByTextKey('shared.header_all').click();
+  }
+
+  public async checkThatAllApplicationsHeaderIsVisible(): Promise<void> {
+    await this.getHeadingByTextKeyAndLevel('dashboard.all_apps', 2).isVisible();
```

Comment:
Same as the link, I find the test more readable when using the native API instead of wrappers. 
```suggestion
    await this.page.getByRole('heading', { name: this.textMock('dashboard.all_apps'), level: 2 }).isVisible();
```